src/hooks/useHaystackRecordTag.ts

```diff
@@ -0,0 +1,107 @@
+import { HDict, HGrid, HRef, HVal } from 'haystack-core'
+import { useEffect, useState } from 'react'
+import { useClient } from '../client'
+import { GridRefreshWatchResult, useWatch } from '../watch'
+
+/**
+ * A Haystack record.
+ */
+export interface Record extends HDict {
+	/**
+	 * Id of the record
+	 */
+	id?: HRef
+}
+
+/**
+ * A hook to use a record tag from an haystack server as react state.
+ *
+ * IMPORTANT: writing is currently only supported with a FIN 5/SkySpark server.
+ *
+ * IMPORTANT: This hooks requires a ClientContext to work. @see https://github.com/j2inn/haystack-nclient
+ *
+ * @example <caption>Example of usage for a precision tag of type Number.</caption>
+ *
+ * const record = useReadByFilter('precision and point and temp and zone and sp').grid[0] // One-shot reading the first record that matches the filter
+ * const [precision, setPrecision, updatedRecord] = useHaystackRecordTag<HNum>(record, "precision") // Using its tag state
+ *
+ * @param originalRecord record dict used to retrieve id and initial value.
+ * @param tagName name of the tag.
+ * @param pollRate poll rate for the value subscription. Expressed in seconds.
+ * @returns Returns the point value, a function to update it and the point dict.
+ */
+export function useHaystackRecordTag<Value extends HVal>(
+	originalRecord?: Record,
+	tagName?: string,
+	pollRate = 5
+): [Optional<Value>, RecordTagWriteFunc<Value> | undefined, Optional<Record>] {
+	const [currentValue, setCurrentValue] = useState<Optional<Value>>()
+	const [tagValue, record] = useReadHaystackRecordTag<Value>(
+		originalRecord,
+		tagName,
+		pollRate
+	)
+	const write = useWriteHaystackRecordTag(originalRecord?.id, tagName)
+	const writeFunc: RecordTagWriteFunc<Value> | undefined = write
+		? async (val) => {
+				const result = await write?.(val)
+				setCurrentValue(val)
+				return result
+		  }
+		: undefined
+
+	useEffect(() => {
+		setCurrentValue(tagValue)
+	}, [tagValue])
+
+	return [currentValue, writeFunc, record]
+}
+
+type RecordGridResult = {
+	grid: HGrid<Record>
+} & GridRefreshWatchResult
+
+type Optional<Value> = Value | undefined | null
+
+function useReadHaystackRecordTag<Value extends HVal>(
+	originalRecord: Optional<Record>,
+	tag = '',
+	pollRate = 5
+): [Optional<Value>, Optional<Record>] {
+	const {
+		grid: [record],
+		isLoading,
+		error,
+	} = useWatch({
+		ids: originalRecord?.id,
+		pollRate: pollRate,
+	}) as RecordGridResult
+
+	if (error) {
+		console.error('useReadHaystackRecord() error:', error)
```

Comment:
Unsure about this error log. Not sure I'd want to see this repeatedly coming up in the console without any control. I'd recommend just removing it.

This also applies to the other places where the same approach is used.